packages/build/tests/mutate_save/tests.js

```diff
@@ -196,6 +196,60 @@ test('--saveConfig saves the configuration changes as netlify.toml', async (t) =
   }
 })
 
+test('--saveConfig does not truncate high amount of redirects', async (t) => {
+  const fixtureDir = `${FIXTURES_DIR}/many_redirects`
+  const fixtureConfigPath = `${fixtureDir}/netlify.toml`
+  const configPath = `${fixtureDir}/test_netlify.toml`
+  await fs.copyFile(fixtureConfigPath, configPath)
+  const { address, stopServer } = await startDeployServer({
+    async onRequest() {
```

Comment:
When `netlifyConfig` is modified, `netlify.toml` is changed on the filesystem right before the deploy call, but restored right after that. Therefore, testing the `netlify.toml`'s contents that's being deployed is not straightforward and requires adding an `onRequest` parameter that's called right in between those two events.